src/test/jtx/impl/AMMTest.cpp

```diff
@@ -0,0 +1,109 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2023 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx/AMM.h>
+#include <test/jtx/AMMTest.h>
+#include <test/jtx/Env.h>
+#include <test/jtx/pay.h>
+
+namespace ripple {
+namespace test {
+namespace jtx {
+
+void
+fund(
+    jtx::Env& env,
+    jtx::Account const& gw,
+    std::vector<jtx::Account> const& accounts,
+    std::vector<STAmount> const& amts,
+    Fund how)
+{
+    fund(env, gw, accounts, 30000 * jtx::dropsPerXRP, amts, how);
+}
+
+void
+fund(
+    jtx::Env& env,
+    std::vector<jtx::Account> const& accounts,
+    STAmount const& xrp,
+    std::vector<STAmount> const& amts,
+    Fund how)
+{
+    for (auto const& account : accounts)
+    {
+        if (how == Fund::All || how == Fund::Acct)
+        {
+            env.fund(xrp, account);
+            env.close();
+        }
+        for (auto const& amt : amts)
+        {
+            env.trust(amt + amt, account);
+            env.close();
+            env(pay(amt.issue().account, account, amt));
+            env.close();
+        }
+    }
```

Comment:
Calling `close()` after every transaction adds a chunk of overhead. You can split this into two loops, `close()` after each one, and get the same results a little bit faster.
```
    for (auto const& account : accounts)
    {
        if (how == Fund::All || how == Fund::Acct)
        {
            env.fund(xrp, account);
        }
    }
    env.close();
    for (auto const& account : accounts)
    {
        for (auto const& amt : amts)
        {
            env.trust(amt + amt, account);
            env(pay(amt.issue().account, account, amt));
        }
    }
    env.close();
```